syntax = "proto3";

package logs;

service ActivityService {
  rpc GetLogs (GetLogsRequest) returns (LogList) {}
  rpc CheckHealth (HealthRequest) returns (HealthResponse) {}
}

message HealthRequest {}
message HealthResponse {
  string status = 1;
}

message GetLogsRequest {
  int32 limit = 1;
}

message LogEntry {
  int32 id = 1;
  int32 user_id = 2; // Changed from string to int32 to match SQLAlchemy model
  string action = 3;
  string timestamp = 4;
  string ip_address = 5;
  string metadata_json = 6;
}

message LogList {
  repeated LogEntry logs = 1;
}
